xarray/conventions/coding.py

```diff
@@ -97,20 +97,22 @@ def _decode_datetime_with_netcdf4(num_dates, units, calendar):
     import netCDF4 as nc4
 
     dates = np.asarray(nc4.num2date(num_dates, units, calendar))
-    if (dates[np.nanargmin(num_dates)].year < 1678 or
-            dates[np.nanargmax(num_dates)].year >= 2262):
-        warnings.warn('Unable to decode time axis into full '
-                      'numpy.datetime64 objects, continuing using dummy '
-                      'netCDF4.datetime objects instead, reason: dates out'
-                      ' of range', RuntimeWarning, stacklevel=3)
-    else:
-        try:
+    if calendar in _STANDARD_CALENDARS:
+        if (dates[np.nanargmin(num_dates)].year < 1678 or
+           dates[np.nanargmax(num_dates)].year >= 2262):
+            warnings.warn(
+                'Unable to decode time axis into full '
+                'numpy.datetime64 objects, continuing using dummy '
+                'netCDF4.datetime objects instead, reason: dates out'
+                ' of range', RuntimeWarning, stacklevel=3)
+        else:
             dates = nctime_to_nptime(dates)
-        except ValueError as e:
-            warnings.warn('Unable to decode time axis into full '
-                          'numpy.datetime64 objects, continuing using '
-                          'dummy netCDF4.datetime objects instead, reason:'
-                          '{0}'.format(e), RuntimeWarning, stacklevel=3)
+    else:
+        warnings.warn('Unable to decode time axis into full numpy.datetime64 '
```

Comment:
I think it makes sense to still warn above (when using a standard calendar, but datetimes are out of the Timestamp-valid range), but should we still warn here (i.e. when using a non-standard calendar)?